models/ssh_key.go

```diff
@@ -600,6 +600,9 @@ type DeployKey struct {
 	Fingerprint string
 	Content     string `xorm:"-"`
 
+	Mode     AccessMode
```

Comment:
It's a little tricky...

`DeployKey` seems to be just a proxy table, with some cached property (like `Name` and `Fingerprint`) of `PublicKey`. Using the same reasoning, I thought that it will be better to have a `Mode` column (like `Name` and `Fingerprint`). 

Another way to do this will be to have retrieved the `Mode` from `PublicKey` (similar to how `Content` is retrieved), but this means an extra query for each access (most probably in `AfterLoad` function, since xorm doesn't that for you...maybe https://github.com/go-xorm/xorm/issues/41 will solve it).

`ReadOnly` is just a property so I don't have to compare `Mode` with `AccessRead`, more like `HasRecentActivity`.